drivers/storage/s3fs/executor/s3fs_executor.go

```diff
@@ -0,0 +1,187 @@
+
+	"github.com/codedellemc/libstorage/api/registry"
+	"github.com/codedellemc/libstorage/api/types"
+
+	"github.com/codedellemc/libstorage/drivers/storage/s3fs"
+	"github.com/codedellemc/libstorage/drivers/storage/s3fs/utils"
+)
+
+// driver is the storage executor for the s3fs storage driver.
+type driver struct {
+	config gofig.Config
+	cmd    string
+	opts   []string
+	szOpts string
+}
+
+func init() {
+	registry.RegisterStorageExecutor(s3fs.Name, newDriver)
+}
+
+func newDriver() types.StorageExecutor {
+	return &driver{}
+}
+
+func (d *driver) Init(ctx types.Context, config gofig.Config) error {
+	d.config = config
+
+	fields := log.Fields{"driver": s3fs.Name}
+	d.cmd = d.config.GetString(s3fs.ConfigS3FSCmd)
+	fields["cmd"] = d.cmd
+
+	if v := d.config.GetStringSlice(s3fs.ConfigS3FSOptions); len(v) > 0 {
+		d.opts = v
+		fields["opts"] = d.opts
+	} else {
+		d.szOpts = d.config.GetString(s3fs.ConfigS3FSOptions)
+		fields["opts"] = d.szOpts
+	}
+
+	ctx.WithFields(fields).Debug("storage executor initialized")
+	return nil
+}
+
+func (d *driver) Name() string {
+	return s3fs.Name
+}
+
+// Supported returns a flag indicating whether or not the platform
+// implementing the executor is valid for the host on which the executor
+// resides.
+func (d *driver) Supported(
+	ctx types.Context,
+	opts types.Store) (bool, error) {
+
+	return gotil.FileExistsInPath(d.cmd), nil
+}
+
+// InstanceID
+func (d *driver) InstanceID(
+	ctx types.Context,
+	opts types.Store) (*types.InstanceID, error) {
+	return utils.InstanceID(ctx, d.config)
+}
+
+// NextDevice returns the next available device.
+func (d *driver) NextDevice(
+	ctx types.Context,
+	opts types.Store) (string, error) {
+	return "", types.ErrNotImplemented
+}
+
+// Return list of local devices
+func (d *driver) LocalDevices(
+	ctx types.Context,
+	opts *types.LocalDevicesOpts) (*types.LocalDevices, error) {
+
+	buckets, err := d.getMountedBuckets(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &types.LocalDevices{Driver: s3fs.Name, DeviceMap: buckets}, nil
+}
+
+func (d *driver) Mount(
+	ctx types.Context,
+	deviceName, mountPoint string,
+	opts *types.DeviceMountOpts) error {
+
+	if mp, ok := d.findMountPoint(ctx, deviceName); ok {
+		ctx.WithFields(log.Fields{
+			"bucket":     deviceName,
+			"mountPoint": mp,
+		}).Debug("bucket is already mounted")
+		if mp == mountPoint {
+			// bucket is mounted to the required target => ok
+			return nil
+		}
+		// bucket is mounted to another target => error
+		return goof.WithFields(goof.Fields{
+			"bucket":      deviceName,
+			"mountPointt": mp,
+		}, "bucket is already mounted")
+	}
+	return d.s3fsMount(ctx, deviceName, mountPoint, opts)
+}
+
+func (d *driver) s3fsMount(
+	ctx types.Context,
+	bucket, mountPoint string,
+	opts *types.DeviceMountOpts) error {
+
+	args := []string{bucket, mountPoint}
+	if len(d.opts) > 0 {
+		for _, o := range d.opts {
+			args = append(args, fmt.Sprintf("-o%s", o))
+		}
+	} else if d.szOpts != "" {
```

Comment:
Hi @codenrhoden,

So, `szOpts` stands for _string of characters terminated by null character_ ... Opts. The `sz` prefix is often found in use by C or C++ programmers to denote a null-terminated string. I will sometimes use it to indicate a variable is a string if there is a similarly named variable of a different type.

The properties in the config file don't really have types per se. It's more how they're interpreted upon being consumed. Thus while it's cleaner to specify the options as an array of `arg=value` or `flag` items, maybe I also want to read it as a string? The above code first attempts to parse the `s3fs.options` value as an array of strings. If it isn't it reads it as a string. That way a user can determine how to put in the options.